pkg/kubelet/runtimeregistry/kuberuntime_registry.go

```diff
@@ -0,0 +1,295 @@
+//
+func buildRuntimeServicesMapFromAgentCommandArgs(remoteRuntimeEndpoints string) (map[string]*RuntimeService, map[string]*ImageService, error) {
+	if remoteRuntimeEndpoints == "" {
+		return nil, nil, fmt.Errorf("runtimeEndpoints is empty")
+	}
+
+	runtimeServices := make(map[string]*RuntimeService)
+	imageServices := make(map[string]*ImageService)
+
+	defer func() {
+		klog.V(4).Infof("runtime services: %v\n", runtimeServices)
+		for _, service := range runtimeServices {
+			klog.V(4).Infof("runtime service name:%s, workloadType:%s, endPointUrl:%s, isDefault:%v, serviceApi:%v",
+				service.Name, service.WorkloadType, service.EndpointUrl, service.IsDefault, service.ServiceApi)
+		}
+		klog.V(4).Infof("image services: %v\n", imageServices)
+		for _, service := range imageServices {
+			klog.V(4).Infof("image service name:%s, workloadType:%s, endPointUrl:%s, isDefault:%v, serviceApi:%v",
+				service.Name, service.WorkloadType, service.EndpointUrl, service.IsDefault, service.ServiceApi)
+		}
+	}()
+
+	endpoints := strings.Split(remoteRuntimeEndpoints, ";")
+
+	// Mostly legacy cluster agent
+	if len(endpoints) == 1 {
+		name, workloadType, endpointUrl, err := parseEndpoint(endpoints[0])
+		if err != nil {
+			return nil, nil, err
+		}
+
+		rs, is, err := getRuntimeAndImageServices(endpointUrl, endpointUrl, metav1.Duration{RuntimeRequestTimeout})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		runtimeServices[name] = &RuntimeService{name,
+			workloadType, endpointUrl, rs, true, true}
+		imageServices[name] = &ImageService{name,
+			workloadType, endpointUrl, is, true}
+
+		return runtimeServices, imageServices, nil
+	}
+
+	firstContainerType := true
+	firstVmType := true
+
+	// Arktos node agent format. multiple runtime services.
+	for _, endpoint := range endpoints {
+		name, workloadType, endpointUrl, err := parseEndpoint(endpoint)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		rs, is, err := getRuntimeAndImageServices(endpointUrl, endpointUrl, metav1.Duration{RuntimeRequestTimeout})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		setDefault := false
+		setPrimry := false
+
+		if workloadType == ContainerWorkloadType && firstContainerType == true ||
+			workloadType == VmworkloadType && firstVmType == true {
+			setDefault = true
+		}
+
+		// Consider first container runtime as primary that must be ready for the node ready condition
+		if workloadType == ContainerWorkloadType && firstContainerType == true {
+			setPrimry = true
+		}
+
+		runtimeServices[name] = &RuntimeService{name,
+			workloadType, endpointUrl, rs, setDefault, setPrimry}
+		imageServices[name] = &ImageService{name,
+			workloadType, endpointUrl, is, setDefault}
+
+		if workloadType == ContainerWorkloadType {
+			firstContainerType = false
+		}
+		if workloadType == VmworkloadType {
+			firstVmType = false
+		}
+	}
+
+	return runtimeServices, imageServices, nil
+}
+
+func parseEndpoint(endpoint string) (string, string, string, error) {
```

Comment:
add unit test if there isn't any already

Reply:
basic UT is in the _test file. this is covered in the buildRuntimeServiceMapFromAgentComandArgs().